src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php

```diff
@@ -569,6 +569,8 @@ public function getValidReadPropertyPaths(): iterable
         yield [(object) [], 'foo?', null];
         yield [(object) ['foo' => (object) ['firstName' => 'Bernhard']], 'foo.bar?', null];
         yield [(object) ['foo' => (object) ['firstName' => 'Bernhard']], 'foo.bar?.baz?', null];
+        yield [(object) ['foo' => null], 'foo?.bar', null];
+        yield [(object) ['foo' => null], 'foo?.bar.baz', null];
```

Comment:
Maybe this test cases would be relevant as well
```php
yield [(object) ['foo' => ['bar' => null], 'foo?.bar?.baz', null];
```

Reply:
good idea (I'd just remove the first `?`):
`yield [(object) ['foo' => ['bar' => null], 'foo.bar?.baz', null];`

Reply:
Added requested tests and also fixes for array access too (like: `yield [(object) ['foo' => ['bar' => null]], 'foo[bar?].baz', null];`).